weepickle-core/src/main/scala/com/rallyhealth/weepickle/v1/core/Util.scala

```diff
@@ -43,16 +43,15 @@ object Util {
 
     intPortion + decPortion
   }
-  def parseLong(cs: CharSequence, start: Int, len: Int): Long = {
+  def parseLong(cs: CharSequence, start: Int, end: Int): Long = {
     // we store the inverse of the positive sum, to ensure we don't
     // incorrectly overflow on Long.MinValue. for positive numbers
     // this inverse sum will be inverted before being returned.
     var inverseSum: Long = 0L
     var inverseSign: Long = -1L
     var i: Int = start
-    val end = start + len
 
-    if ((start | len | (cs.length - end) | end) < 0) throw new IndexOutOfBoundsException
+    if ((start | end | end - start | cs.length - end) < 0) throw new IndexOutOfBoundsException
```

Comment:
Wow, hadn't noticed this bit before.  I can see the optimization, but boy I'm unused to thinking in terms of this sort of bit-twiddling these days.

Reply:
cats is pretty far removed from it. :)

I stole the approach from the java standard library where it's used heavily in `java.io.OutputStream` classes, etc.